### PR TITLE
feat: add monitor

### PR DESCRIPTION
**Pull request by Redish101**
## Description

Migrate monitor middleware from fiber v2 to v3. See at [gofiber/fiber#3125](https://github.com/gofiber/fiber/pull/3125)

## Summary by CodeRabbit

- **New Features**
	- Introduced a structured template for automated release notes management in GitHub.
	- Added a GitHub Actions workflow to automate the drafting of release notes.
	- Implemented a web-based monitoring interface for system metrics with real-time updates.
	- Developed a middleware for monitoring system and process statistics, providing JSON and HTML responses.

- **Documentation**
	- Added a comprehensive README file for the "monitor" component to enhance user guidance and documentation structure.

### CODE REVIEW

File: monitor/README.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+| APIOnly    | `bool`                    | Whether the service should expose only the montioring API.                           | `false`                                                                     |
+| Next       | `func(c *fiber.Ctx) bool` | Define a function to add custom fields.                                              | `nil`                                                                       |
+| CustomHead | `string`                  | Custom HTML conde to Head Section(Before End).                                       | `empty`                                                                     |
+| FontURL    | `string`                  | FontURL for specilt font resource path or URL. also you can use relative path.       | `https://fonts.googleapis.com/css2?family=Roboto:wght@400;900&display=swap` |
+| ChartJsURL | `string`                  | ChartJsURL for specilt chartjs library, path or URL, also you can use relative path. | `https://cdn.jsdelivr.net/npm/chart.js@2.9/dist/Chart.bundle.min.js`        |
+
+> Because jsdelivr lost their ICP license, so chinese users maybe use other CDNs to load ChartJs library.
```
Comment (gaby) [inline monitor/README.md:39]: This comment is not relevant here, it's up to the user to handle their CDN

File: monitor/go.mod [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module github.com/gofiber/contrib/monitor
+
+go 1.22.6
```
Comment (gaby) [inline monitor/go.mod:3]: This should be `1.22`, without the patch

File: .github/workflows/release-drafter-monitor.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Release Drafter Casbin
```
Comment (gaby) [inline .github/workflows/release-drafter-monitor.yml:1]: Monitor* instead of Casbin

File: .github/workflows/release-drafter-monitor.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+on:
+    push:
+        # branches to consider in the event; optional, defaults to all
+        branches:
+            - master
+            - main
+        paths:
+            - 'monitor/**'
+jobs:
+    draft_release_casbin:
```
Comment (gaby) [inline .github/workflows/release-drafter-monitor.yml:11]: Monitor instead of Casbin

File: .github/release-drafter-monitor.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            - 'patch'
+            - '📒 Documentation'
+            - '☢️ Bug'
+            - '🤖 Dependencies'
+            - '🧹 Updates'
+    default: patch
+template: |
+    $CHANGES
+
+    **Full Changelog**: https://github.com/$OWNER/$REPOSITORY/compare/$PREVIOUS_TAG...casbin/v$RESOLVED_VERSION
```
Comment (gaby) [inline .github/release-drafter-monitor.yml:48]: Monitor instead of Casbin

File: monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    "github.com/gofiber/fiber/v3"
+    "github.com/gofiber/contrib/monitor"
+)
+
+func main() {
+    app := fiber.New()
+
+    app.Use("/monitor", monitor.New())
+
```
Comment (gaby) [inline monitor/README.md:57]: Remove extra newline here

File: monitor/monitor.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+			for {
+				time.Sleep(cfg.Refresh)
+
+				updateStatistics(p, numcpu)
+			}
+		}()
+	})
+
+	// Return new handler
+	//nolint:errcheck // Ignore the type-assertion errors
```
Comment (gaby) [inline monitor/monitor.go:76]: I think this "nolint" can be removed

**Review by gaby — CHANGES_REQUESTED** [review]

**Comment by Redish101** [comment]
@gaby Sorry, these changes should be committed to the v3-beta branch. Please refer to PR #1172

**Pull request closed without merging**